pkg/services/folder/folderimpl/unifiedstore.go

```diff
@@ -447,6 +445,43 @@ func (ss *FolderUnifiedStoreImpl) CountInOrg(ctx context.Context, orgID int64) (
 	return resp.Stats[0].Count, nil
 }
 
+func (ss *FolderUnifiedStoreImpl) list(ctx context.Context, orgID int64, opts v1.ListOptions) (*unstructured.UnstructuredList, error) {
+	var allItems []unstructured.Unstructured
+
+	listOpts := opts.DeepCopy()
+
+	if listOpts.Limit == 0 {
+		listOpts.Limit = folderListLimit
+	}
+
+	for {
+		out, err := ss.k8sclient.List(ctx, orgID, *listOpts)
+		if err != nil {
+			return nil, err
+		}
+
+		if out != nil && len(out.Items) > 0 {
+			allItems = append(allItems, out.Items...)
```

Comment:
also, I think that it's possible to trespass the limit we defined in opts.Limit

for example, opts.Limit=15,  allItems contains 10 items, then we get 10 more, resulting in 20 items. 
We wont be respecting the limit in this case.

Reply:
ah!! I see that you cut the list at the end of this method!